sim/common/cata/other_effects.go

```diff
@@ -1435,6 +1436,7 @@ func RegisterIgniteEffect(unit *core.Unit, config IgniteConfig) *core.Spell {
 		ActionID:         config.ActionID,
 		SpellSchool:      core.SpellSchoolFire,
 		ProcMask:         core.ProcMaskSpellProc,
+		ClassSpellMask:   config.ClassSpellMask,
```

Comment:
I added this because sometimes (like in my case) you'd want to actually give the Ignite spell a mask